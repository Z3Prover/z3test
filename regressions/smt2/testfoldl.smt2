(push)
(declare-fun f (Int Int) Int)
(declare-const il (Seq Int))
(assert (= (seq.foldl f 0 il) 5))
(check-sat)
(pop)

(push)
(declare-const il (Seq Int))
(declare-const F (Array Bool Int Bool))
(assert (= (seq.foldl F true il) true))
(assert (> (seq.len il) 0))
(assert (not (= F ((as const (Array Bool Int Bool)) true))))
(check-sat)
(pop)


(push)
(declare-fun f (Int Int Int) Int)
(declare-const il (Seq Int))
(assert (= (seq.foldli f 0 0 il) 5))
(check-sat)
(pop)

(push)
(declare-const il (Seq Int))
(declare-const F (Array Int Bool Int Bool))
(assert (= (seq.foldli F 5 true il) true))
(assert (> (seq.len il) 0))
(assert (not (= F ((as const (Array Int Bool Int Bool)) true))))
(check-sat)
(pop)


