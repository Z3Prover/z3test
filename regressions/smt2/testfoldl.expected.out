sat
sat
sat
sat
(seq.fold_left f (seq.fold_left f 2 il_1) il_2)
(select f 3 i)
(seq.fold_leftli g (+ 2 (seq.len il_1)) (seq.fold_leftli g 2 false il_1) il_2)
(select g 3 true i)
2
true
